sdk/resource/env.go

```diff
@@ -88,7 +90,11 @@ func constructOTResources(s string) (*Resource, error) {
 			invalid = append(invalid, p)
 			continue
 		}
-		k, v := strings.TrimSpace(field[0]), strings.TrimSpace(field[1])
+		k := strings.TrimSpace(field[0])
+		v, err := url.QueryUnescape(strings.TrimSpace(field[1]))
+		if err != nil {
+			otel.Handle(err)
+		}
 		attrs = append(attrs, attribute.String(k, v))
```

Comment:
Oh, I see it now :facepalm: 

Thanks for explanation and sorry for the confusion :sweat_smile: 

For testing the call to the error handler, I was thinking of doing something like [this](https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/metric/controller/basic/push_test.go#L41-L65), but I'm not sure if there's a more standardized way of doing it rather than copying this handler over.